### ISSUE TITLE
[Feature Request / Idea]: Use XDG Base Directory Specification by default

### ISSUE THREAD

**Issue by JacobTravers** [rusty-ferris-club/shellclear#65]
### Suggestion / Feature Request

The `shellclear/` and `.shellclear/` directories unnecessarily pollute users' home directory.

The [XDGBDS](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) defines where these files should be placed with environment variables in order to avoid this.

> The XDG Base Directory Specification is based on the following concepts:
>
>   - There is a single base directory relative to which user-specific data files should be written. This directory is defined by the environment variable `$XDG_DATA_HOME`.
>
>   - There is a single base directory relative to which user-specific configuration files should be written. This directory is defined by the environment variable `$XDG_CONFIG_HOME`.
>
>   - There is a single base directory relative to which user-specific state data should be written. This directory is defined by the environment variable `$XDG_STATE_HOME`.
>
>   - There is a single base directory relative to which user-specific executable files may be written.
>
>   - There is a set of preference ordered base directories relative to which data files should be searched. This set of directories is defined by the environment variable `$XDG_DATA_DIRS`.
>
>   - There is a set of preference ordered base directories relative to which configuration files should be searched. This set of directories is defined by the environment variable `$XDG_CONFIG_DIRS`.
>
>   - There is a single base directory relative to which user-specific non-essential (cached) data should be written. This directory is defined by the environment variable `$XDG_CACHE_HOME`.
>
>   - There is a single base directory relative to which user-specific runtime files and other file objects should be placed. This directory is defined by the environment variable `$XDG_RUNTIME_DIR`.
>
> All paths set in these environment variables must be absolute. If an implementation encounters a relative path in any of these variables it should consider the path invalid and ignore it.

**Comment by kaplanelad** [comment]
Hey @JacobTravers, thanks for reporting this issue.

Just to make sure that i understand, you suggestion is to manage the `.shellclear` in $XDG_CACHE_HOME instead of $HOME ?

**Comment by JacobTravers** [comment]
Correct, so it should check first if `$XDG_CACHE_HOME` exists, if so it would use `$XDG_CACHE_HOME/shellclear` and if not it would fall back to `.shellclear`.

As for `shellclear/`, I might be mistaken but I believe there was an empty `shellclear/` directory created in `$HOME` when I installed this with `cargo install --git`, but I'm not sure what data would have been there or how it would fit into the XDGBDS. It may have been a typo.

**Comment by boaz-quotient** [comment]
@kaplanelad can i work on it?

**Comment by kaplanelad** [comment]
Sure @boaz-quotient Thank you.
In shellclear i'm using [dirs](https://crates.io/crates/dirs) crates. in this lib you have the function [config_dir](https://github.com/dirs-dev/dirs-rs/blob/main/src/lib.rs#L98) that already taking the $XDG_CONFIG_HOME with a default.

Please note that we need to support the old version, which means if the user already has the config file in $HOME path, we need to take it first

Thanks again
